### PR TITLE
Issue quota & billing requests against the provided project

### PR DESCRIPTION
**Pull request by sethvargo**
This is important for cross-project requests. If you are authenticated as `project-one` but set the `project_id` as `project-two`, the API quota will still be billed against the project that was authenticated, not the specified `project_id`. This fixes that.

### CODE REVIEW

**Comment by sethvargo** [comment]
Hmm interesting. I think we might need to discuss this.

**Comment by matthijsvanderloos** [comment]
I am running into this issue as well.

Also tried with gcloud and explicitly setting the billing project (`gcloud container clusters get-credentials YYY --zone ZZZ --project XXX --billing-project WWW`), but this results in:

`ERROR: (gcloud.container.clusters.get-credentials) ResponseError: code=403, message=Caller does not have required permission to use project WWW. Grant the caller the roles/serviceusage.serviceUsageConsumer role, or a custom role with the serviceusage.services.use permission, by visiting https://console.developers.google.com/iam-admin/iam/project?project=WWW and then retry. Propagation of the new permission may take a few minutes.`

So the same as in the PR checks. Is there any update?

File: README.md [hunk 1]
```diff
@@ -91,6 +91,11 @@ jobs:
     for generating the Connect Gateway endpoint. This only applies if
     "use_connect_gateway" is true. Defaults to auto discovery if empty.
 
+-   `quota_project_id` - (Optional) Project ID from which to pull quota. The
```
Comment (verbanicm) [inline README.md:94]: Should we call this `billing_project_id` to align with gcloud?

https://cloud.google.com/sdk/gcloud/reference#--billing-project

Comment (sethvargo) [reply README.md:94]: There's no consistent name for it:

- billing project
- quota project
- user project
- user quota project

In this case, the API calls to GKE aren't billed (like KMS), so it's not really a "billing" project; it's really just about where quota is allocated.

**Review by verbanicm — APPROVED** [review]

**Pull request merged**